tests/test_swin_unetr.py

```diff
@@ -38,23 +45,25 @@
                 for img_size in ((64, 32, 192), (96, 32)):
                     for feature_size in [12]:
                         for norm_name in ["instance"]:
-                            test_case = [
-                                {
-                                    "spatial_dims": len(img_size),
-                                    "in_channels": in_channels,
-                                    "out_channels": out_channels,
-                                    "img_size": img_size,
-                                    "feature_size": feature_size,
-                                    "depths": depth,
-                                    "norm_name": norm_name,
-                                    "attn_drop_rate": attn_drop_rate,
-                                    "downsample": test_merging_mode[case_idx % 4],
-                                },
-                                (2, in_channels, *img_size),
-                                (2, out_channels, *img_size),
-                            ]
-                            case_idx += 1
-                            TEST_CASE_SWIN_UNETR.append(test_case)
+                            for use_checkpoint in [True, False]:
```

Comment:
Thanks! Have updated it in the latest commit.

Reply:
I think you've changed these tests back to missing the problem of not testing with `use_checkpoint == True`. What I had in mind was more of having a variable before the for loops like `checkpoint_vals = [True, False] if pytorch_after(1,11) else [False]` and then this for loop becomes `for use_checkpoint in checkpoint_vals:`. We wouldn't need the check for `SkipIfBeforePyTorchVersion` in this case in any of the tests since the issue with older versions would be filtered out in this way.

Reply:
I tested the `use_checkpoint == True` in this case. https://github.com/Project-MONAI/MONAI/blob/163bdd6e5049681fdb13e9bcda9de91e738f106a/tests/test_swin_unetr.py#L68

But your proposal could also be fine, thanks! I have updated it in this commit https://github.com/Project-MONAI/MONAI/pull/7266/commits/6ccbe7185c06c8c6ebc126cbc55206e046240c20